tests/io/test_pyarrow_visitor.py

```diff
@@ -583,46 +584,79 @@ def test_pyarrow_schema_to_schema_fresh_ids_nested_schema(
     assert visit_pyarrow(pyarrow_schema_nested_without_ids, _ConvertToIcebergWithoutIDs()) == iceberg_schema_nested_no_ids
 
 
-def test_pyarrow_schema_ensure_large_types(pyarrow_schema_nested_without_ids: pa.Schema) -> None:
-    expected_schema = pa.schema(
+def test_pyarrow_schema_unsupported_type() -> None:
```

Comment:
we probably dont need the whole setup here. 
Testing `pyarrow_to_schema` is enough, for example:
```
unsupported_field = pa.field("latitude", pa.decimal256(20, 26), nullable=False)
schema = pa.schema([unsupported_field])
pyarrow_to_schema(schema) # will raise here
```